### ISSUE TITLE
FSM: stop/start behaviour clarification 

### ISSUE THREAD

**Issue by hjrnunes** [ingesolvoll/kee-frame#96]
Hi.

When stopping and starting FSMs, shouldn't the FSM state be restarted according to the `:start` directive?

*Example*:

```
(require '[kee-frame.fsm.alpha :as fsm])

(def my-http-request-fsm
   {:id    :my-http-request
    :start ::loading
    :stop  ::loaded
    :fsm   {::loading        {[::fsm/on-enter]      {:dispatch [[:contact-the-server]]}
                              [:server-responded]   {:to ::loaded}
                              [:default-on-failure] {:to ::loading-failed}}
            ::loading-failed {[::fsm/timeout 10000] {:to ::loading}}}})
```

If this FSM is stopped and I dispatch `::fsm/start my-http-request-fsm` to start it again, shouldn't `:state` become `::loading`? 

I'm seeking to essentially reset the FSM via `:start`/`:stop`, and if I dispatch `:stop` and then `:start`, the `:state` remains whatever it was on when `:stop` was dispatched.

I was expecting state to be reset on `:stop`, but maybe I'm missing something?

Thanks

**Comment by brjann** [comment]
I had a conversation about this with @ingesolvoll on Slack. He suggested that the `::loaded` state should not be the `:stop` state, but rather a normal state that awaits a restart (e.g., `:restart`) event, which brings the fsm into the `::loading` state again. You then dispatch the :restart event in your controller's `:start` function _and_ you return the fsm. So if it's not already started, the fsm handler takes care of that, otherwise the `:restart` will restart it.

I tried it out and it works fine, so I don't use the `:stop` configuration.

But I have been thinking about what the purpose of the `:stop` configuration is and how it is supposed to work. Is it a final stopped state and the fsm will never start again? Used to detach the fsm from listening to state change events?

And what is the function of the `::fsm/stop` event? The documentation states that 

> FSMs can be started and stopped like this:
> 
>     (rf/dispatch [::fsm/start my-http-request-fsm])
>     (rf/dispatch [::fsm/stop my-http-request-fsm])

and also that when a controller returns an fsm in its :start function, "This FSM will be started and stopped by the controller start/stop lifecycle"

I may misinterpret this, but it sounds like something that I would not want the controller to do. If my fsm is in some state that needs a proper teardown, I don't want it to be forced to enter its `:stop` state. So I actually dispatch a `:leave` even in the controller's `:stop` function that triggers a transition in the fsm.

It would be great to hear @ingesolvoll's thoughts on how the stopping of fsm is supposed!

Thanks for the great work on kee-frame!

**Comment by hjrnunes** [comment]
> I had a conversation about this with @ingesolvoll on Slack, _etc._

Yes. That makes sense. I ended up rolling my own version of `with-fsm` and `render` that will dispatch something to remove the fsm state from the app-db, because I almost always want that. When I don't, I use kee-frame's.

> And what is the function of the ::fsm/stop event?
> Used to detach the fsm from listening to state change events?

I think this is it.

> Thanks for the great work on kee-frame!

Seconded!
